packages/solana/lib/wallet/SolanaWalletProvider.ts

```diff
@@ -83,38 +90,54 @@ export class SolanaWalletProvider extends Wallet<Connection, Promise<Keypair>> {
         return Buffer.from(signature).toString('hex');
     }
 
-    public async sendTransaction(txRequest: TransactionRequest): Promise<Transaction> {
-        const to = new PublicKey(txRequest.to.toString());
+    public async sendTransaction(txRequest: SolanaTxRequest): Promise<Transaction> {
+        let transaction: SolTransaction;
 
-        let instruction: TransactionInstruction;
-        // Handle ERC20 Transactions
-        if (txRequest.asset && !txRequest.asset.isNative) {
-            const contractAddress = new PublicKey(txRequest.asset.contractAddress);
-            const fromTokenAccount = await getAssociatedTokenAddress(contractAddress, this._signer.publicKey);
+        const latestBlockhash = await retry(async () => this.chainProvider.getProvider().getLatestBlockhash('confirmed'));
 
-            try {
-                await createAccount(this.chainProvider.getProvider(), this._signer, contractAddress, to);
-            } catch (err) {
-                logger.debug(`Error creating account`, err);
+        // Handle already builded transactions that are passed from outside - Jupiter for example
+        if (txRequest.transaction) {
+            transaction = txRequest.transaction;
+            transaction.recentBlockhash = latestBlockhash.blockhash;
+        } else {
+            let instruction: TransactionInstruction;
+            const to = new PublicKey(txRequest.to.toString());
+
+            // Handle ERC20 Transactions
+            if (txRequest.asset && !txRequest.asset.isNative) {
+                const contractAddress = new PublicKey(txRequest.asset.contractAddress);
+
+                const [fromTokenAccount, toTokenAccount] = await Promise.all([
+                    getAssociatedTokenAddress(contractAddress, this._signer.publicKey),
+                    getAssociatedTokenAddress(contractAddress, to),
+                ]);
+
+                try {
+                    await getAccount(this.chainProvider.getProvider(), toTokenAccount);
+                } catch (err) {
+                    if (err instanceof TokenAccountNotFoundError) {
+                        await createAccount(this.chainProvider.getProvider(), this._signer, contractAddress, to);
```

Comment:
Does this send an extra transaction from the user account in certain situations? 
If this is the case, it is happening without the user realising? And that feels wrong

Reply:
I agree but thats how the solana in general works, even phantom wallet behave at the same way, they are creating token accounts for you under the hood

Reply:
Also transactions have fixed fee which is `00000000.5 SOL` which is very low low amount in cent value, thats why i assume they are doing this under the hood